### PR TITLE
Let i18n:ignore ignore all translatable strings in its scope.

### PR DESCRIPTION
**Pull request by mauritsvanrees**
Main use case: add i18n:ignore in the top level tag of a template to let the find-untranslated command ignore the complete template. This is useful when you know that all managers or editors can read English, so need no translation.
The alternative would be to add an `--exclude` parameter to the `find-untranslated` command.

This PR builds on PR #108 which cleans up the code.  So you may want to review that first.

### CODE REVIEW

File: src/i18ndude/untranslated.py [hunk 1]
```diff
@@ -213,36 +218,38 @@ def endDocument(self):
     def startElement(self, tag, attrs):
         self._history.append([tag, attrs, ""])
 
-        attr_validator(tag, attrs, self.log)
-
         if "i18n:translate" in attrs.keys():
             self._i18nlevel += 1
         elif self._i18nlevel != 0:
             self._i18nlevel += 1
+        if IGNORE_UNTRANSLATED in attrs.keys():
```
Comment (ale-rt) [inline src/i18ndude/untranslated.py:225]:
```suggestion
        if IGNORE_UNTRANSLATED in attrs:
```

File: src/i18ndude/untranslated.py [hunk 2]
```diff
@@ -213,36 +218,38 @@ def endDocument(self):
...
+            severity = _severity(tag, attrs) or ""
+            if severity and not CHAMELEON_SUBST.match(data):
+                self.log(
+                    "i18n:translate missing for this:\n" '"""\n%s\n"""' % (data,),
+                    severity,
+                )
 
         if self._i18nlevel != 0:
             self._i18nlevel -= 1
+        if IGNORE_UNTRANSLATED in attrs.keys():
```
Comment (ale-rt) [inline src/i18ndude/untranslated.py:250]:
```suggestion
        if IGNORE_UNTRANSLATED in attrs:
```

**Review by ale-rt — APPROVED** [review]
LGTM! Suggestions can be gonored

**Review by gforcada — APPROVED** [review]
LGTM!

I was missing this feature a few weeks ago, where I ended up filling a template with `i18n:ignore` 😅

**Comment by mauritsvanrees** [comment]
> I was missing this feature a few weeks ago, where I ended up filling a template with `i18n:ignore` 😅

That is exactly what I was doing yesterday, and I decided to create this PR instead. :-)

**Pull request merged**